### PR TITLE
SYCL: Migrate away from deprecated ggml_tensor->backend

### PR DESCRIPTION
**Pull request by qnixsynapse**
This should have been done way earlier as `ggml_tensor->backend` has been deprecated for a very long time.

There are some doubts in this, and thus have added comments which I will remove after discussing with the collaborators.
So far, backend test ops (for single GPU) are passing with this change.

cc: @airMeng @NeoZhangJianyu @abhilash1910 @Rbiessy

Also, integrated with GGML_LOG for debug logs and remove backend specific logging system.
With new log, it will be better to debug, for example:
```
call ggml_sycl_rms_norm                         // LLAMA model's attention part, currently uses eager attention
call ggml_sycl_rms_norm done
call ggml_sycl_mul
call ggml_sycl_mul done
[SYCL]: call ggml_sycl_mul_mat
call ggml_sycl_rope                                   
call ggml_sycl_rope done
[SYCL]: call ggml_sycl_mul_mat
call ggml_sycl_rope                                   
[SYCL]: call ggml_sycl_mul_mat
[SYCL]: call ggml_sycl_mul_mat
call ggml_sycl_soft_max
call ggml_sycl_soft_max done
[SYCL]: call ggml_sycl_mul_mat
call ggml_sycl_dup
call ggml_sycl_dup done
[SYCL]: call ggml_sycl_mul_mat
call ggml_sycl_add
call ggml_sycl_add done
------------------------------------------------------
call ggml_sycl_rms_norm                                                          // LLAMA Feed forward MLP SWIGLU
call ggml_sycl_rms_norm done
call ggml_sycl_mul
call ggml_sycl_mul done
[SYCL]: call ggml_sycl_mul_mat
call ggml_sycl_silu
call ggml_sycl_silu done
[SYCL]: call ggml_sycl_mul_mat
call ggml_sycl_mul
call ggml_sycl_mul done
[SYCL]: call ggml_sycl_mul_mat
call ggml_sycl_add
call ggml_sycl_add done
```
Wish there was a level to control the verbosity using environmental variables of the debug log levels rather than passing command line arguments. This is reverted for now..

### CODE REVIEW

**Review by slaren — APPROVED** [review]

**Comment by Rbiessy** [comment]
Thanks for the PR. FYI most of us are on holiday so we may not be able to review until next month. If this is not urgent please give us some time to review it.

File: ggml/src/ggml-sycl/ggml-sycl.cpp
Comment (abhilash1910) [inline ggml/src/ggml-sycl/ggml-sycl.cpp:149]: If we are removing this, is there any other addition where sycl env is called for logging?

Comment (NeoZhangJianyu) [reply ggml/src/ggml-sycl/ggml-sycl.cpp:149]: Same comment.
I suggest keeping this function.

**Review by NeoZhangJianyu — COMMENTED** [review]
Thank your for the improvement.
But I don't think it bring benefit to SYCL backend user:
1.
`GGML_SYCL_DEBUG` is used to debug the performance and error in SYCL backend only.  
It can be opened by environment in user running time without rebuild the source code.
Looks like `GGML_LOG_DEBUG`  need to rebuild the source code after define macro.
It limits the online trouble shooting.

The `GGML_SYCL_DEBUG` function is more powerful than `GGML_LOG_DEBUG`.
We can't reduce the function for unify the code.
Customer should be first. :)

2.
Replace it by common `GGML_LOG_DEBUG` log function, will mix the logs of common code and SYCL backend.
In performance test, there will be more log info.

File: ggml/src/ggml-sycl/ggml-sycl.cpp
Comment (qnixsynapse) [reply ggml/src/ggml-sycl/ggml-sycl.cpp:149]: According to PR #9709 , the backend specific logging system was supposed to be replaced with a common logging system. Although I agree that this will enable all debug logs if `--log-verbose` cmdline argument is passed or enabled by default in test-backend-ops.

cc @slaren I think best here is to enable them only when something like `GGML_BACKEND_DEBUG=1` is enabled.

**Review by NeoZhangJianyu — APPROVED** [review]

File: ggml/src/ggml-sycl/ggml-sycl.cpp
Comment (abhilash1910) [reply ggml/src/ggml-sycl/ggml-sycl.cpp:2354]: Agree with this.

File: ggml/src/ggml-sycl/ggml-sycl.cpp
Comment (abhilash1910) [reply ggml/src/ggml-sycl/ggml-sycl.cpp]: Sounds good. for platform benchmark purposes it makes sense to have the platform specific debug on.

File: ggml/src/ggml-sycl/ggml-sycl.cpp
Comment (qnixsynapse) [reply ggml/src/ggml-sycl/ggml-sycl.cpp:2354]: As you already know that the previous backend path which used int(enum) is deprecated for a long time:

https://github.com/ggerganov/llama.cpp/blob/152610eda91217ac409342cd976d05f5114ad39f/ggml/include/ggml.h#L590

I reused the already existed buffer implementation.
Also, I didn't notice any noticible slowdowns. If you do, please share the results.

**Review by Rbiessy — APPROVED** [review]
I have no concern with the changes here but I'll let @NeoZhangJianyu merge it if he's happy with the recent changes.

**Comment by qnixsynapse** [comment]
@Rbiessy Thank you. If only I could get the result of test on a multi GPU/RPC setup before merging, it would be nice since I couldn't able to do that due to lack of hardware.

**Comment by Rbiessy** [comment]
We also don't test multi-GPU or RPC setup at Codeplay so far. I tested this with a few models on A100 and found no regression.

**Pull request merged**
